src/Moq.Analyzers/EnumerableExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Moq.Analyzers;
+
+internal static class EnumerableExtensions
+{
+    /// <inheritdoc cref="SingleWhenOnly{TSource}(IEnumerable{TSource}, Func{TSource, bool})"/>
+    public static TSource? SingleWhenOnly<TSource>(this IEnumerable<TSource> source)
+    {
+        return source.SingleWhenOnly(_ => true);
+    }
+
+    /// <summary>
+    /// Returns the only element of a sequence that satisfies a specified condition,
+    /// and returns a default value if no such element exists or there are multiple matches.
+    /// </summary>
+    /// <typeparam name="TSource">The type of the <paramref name="source"/> collection.`</typeparam>
+    /// <param name="source">The collection to enumerate.</param>
+    /// <param name="predicate">A function to test each element for a condition.</param>
+    /// <returns><see langword="true"/> if there is only one element in the collection; <see langword="false"/> otherwise.</returns>
```

Comment:
The returns doc is wrong, you return an instance of `TSource` or `default`

Reply:
Whoops! Fixed!